plugins/transfer/transfer.plugin.zsh

```diff
@@ -0,0 +1,25 @@
+# From https://transfer.sh/
+# Modified by Gabriel Kaam <gabriel@kaam.fr>
+
+function transfer() {
+  if [[ $# -eq 0 ]]; then
+    cat<<EOF
+No arguments specified. Usage:
+transfer /tmp/test.md
+cat /tmp/test.md | transfer test.md
+EOF
+    return 1
+  fi
+
+  basefile=$(omz_urlencode "${1:t}")
+
+  if tty -s; then
```

Comment:
The only difference between the two branches of this `if` are the infile. Maybe it could be factored out, and construct the `curl` command in one place? Could help readability, and make it easier to add additional options later on. 

And `tty -s` seems like the wrong test to do here. You might be calling `transfer` inside a script or function which itself is hooked up to a pipeline, but still want to use the form where you give it a filename instead of using `stdin`. Alternately, if you allow the no-arg `stdin` form from the terminal as well, then a user could run transfer, and type or paste stuff from the keyboard to go in to the uploaded file.

How about something like:

```
# Default to reading stdin
local infile="$1"
basefile=$(omz_urlencode "${infile:t}")
if [[ -z $infile ]]; then
   infile="-"
fi

url=$(curl --progress-bar --upload-file "$infile" "https://transfer.sh/$basefile")
```

Or, if you wanted to be more friendly to beginning users, when you get no infile specified, you could then check to see if you're in a terminal, and if so give the user an error message, and require them to explicitly pass "-" as the infile if they really want to use `stdin` from the terminal. Otherwise, you can get the case where you're calling `curl --upload-file "$1"` and `$1` is empty, which will look like a normal web page hit and end up dumping out the `transfer.sh` usage instructions instead of an upload result URL, like this.

```
$ curl --progress-bar --upload-file "" "https://transfer.sh/"
transfer.sh: Easy file sharing from the command line
===
made with <3 by DutchCoders

Upload:
$ curl --upload-file ./hello.txt https://transfer.sh/hello.txt

Encrypt & upload:
```